### DISCUSSION TITLE
74191 Circuit : cant analyze

### DISCUSSION THREAD

**Discussion by GlennFR** [hneemann/Digital#1294] [Q&A] [answered]
Hello, I'm very new to Digital - so I've probably made a silly mistake or forgotten something.  I want to analyze a simple circuit with just a 74191 U/D counter.  When I try to Analyze it, I get a message saying it cant analyze a JKFlipflop (I suppose its an internal element of the 74191 definition).
![74191_Analysis-Error](https://github.com/hneemann/Digital/assets/56517031/f29f4031-d9f3-41b6-a4df-8fe2965652c4)
If I press the RUN button, the simulation seems to work as it should.  Can someone help me, please?

**Comment by hneemann** [comment, marked as answer]
What do you want to achieve with the analysis?
As the error message says, an asynchronous FF cannot be analyzed in order to build a Mealy or Moore finite-state machine, for example. And the 74191 contains four asynchronous FFs. If you use one of the built-in counters, the analysis works.

**Reply by GlennFR** [reply]
Oh, I see.  I thought that it was an error inside the "device", not something that isn't possible...  Perhaps something could be added to the help/documentation?  I never studied CPLD development before and until recently didn't even know what the Mealy machine or Moore machine were.  Thankyou.
